File: Quiz_11/AKSHCHAUHAN_500120187/Quiz_11_Debugging_Question_2.c
#include <stdio.h>
#include <stdlib.h>
struct TreeNode {
    int val;
    struct TreeNode* left;
    struct TreeNode* right;
};
struct TreeNode* newNode(int val) {
    struct TreeNode* node = (struct TreeNode*)malloc(sizeof(struct TreeNode));
    node->val = val;
    node->left = node->right = NULL;
    return node;
}
int convert_to_sum_tree(struct TreeNode* root) {
    if (root == NULL) return 0;
    int old_val = root->val;
    int left_sum = convert_to_sum_tree(root->left);
    int right_sum = convert_to_sum_tree(root->right);
    root->val = left_sum + right_sum;
    return root->val + old_val;
}

void print_inorder(struct TreeNode* root) {
    if (root == NULL) return;
    print_inorder(root->left);
    printf("%d ", root->val);
    print_inorder(root->right);
}

int main() {
    struct TreeNode* root = newNode(1);
    root->left = newNode(2);
    root->right = newNode(3);
    root->left->left = newNode(4);
    root->left->right = newNode(5);
    printf("Original tree (inorder): ");
    print_inorder(root);
    printf("\n");
    convert_to_sum_tree(root);
    printf("After conversion (inorder): ");
    print_inorder(root);
    printf("\n");
    return 0;
}

File: Quiz_11/Ayush_Mishra_590014844/Quiz_11_Debugging_Question_2.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def convert_to_sum_tree(root):
    if root is None:
        return 0

    left_sum = convert_to_sum_tree(root.left)
    right_sum = convert_to_sum_tree(root.right)

    old_val = root.val

    root.val = left_sum + right_sum

    return root.val + old_val

def print_inorder(root):
    if root:
        print_inorder(root.left)
        print(root.val, end=" ")
        print_inorder(root.right)

# Test case:
#     1
#    / \
#   2   3
#  / \
# 4   5
root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)
root.left.left = TreeNode(4)
root.left.right = TreeNode(5)

print("Original tree (inorder):")
print_inorder(root)
print()

convert_to_sum_tree(root)

print("After conversion to Sum Tree (inorder):")
print_inorder(root)
print()
